src/plugins/plugin.gridlines.js

```diff
@@ -0,0 +1,182 @@
+		} else {
+			lineValue -= (lineValue - scale.getPixelForTick(index - 1)) / 2;
+		}
+	}
+	return lineValue;
+}
+
+module.exports = function() {
+	function getGridLine(chart, scale, position, tickIndex) {
+		var chartArea = chart.chartArea;
+		var scaleOptions = scale.options;
+		var gridLines = scaleOptions.gridLines;
+
+		var lineWidth, lineColor, borderDash, borderDashOffset;
+
+		if (tickIndex !== undefined && tickIndex === scale.zeroLineIndex) {
+			lineWidth = gridLines.zeroLineWidth;
+			lineColor = gridLines.zeroLineColor;
+			borderDash = gridLines.zeroLineBorderDash;
+			borderDashOffset = gridLines.zeroLineBorderDashOffset;
+		} else {
+			lineWidth = helpers.getValueAtIndexOrDefault(gridLines.lineWidth, tickIndex);
+			lineColor = helpers.getValueAtIndexOrDefault(gridLines.color, tickIndex);
+			borderDash = helpers.getValueOrDefault(gridLines.borderDash, defaults.global.borderDash);
+			borderDashOffset = helpers.getValueOrDefault(gridLines.borderDashOffset, defaults.global.borderDashOffset);
+		}
+
+		var x1, x2, y1, y2;
+
+		if (scale.isHorizontal()) {
+			x1 = x2 = position;
+			y1 = chartArea.top;
+			y2 = chartArea.bottom;
+		} else {
+			x1 = chartArea.left;
+			x2 = chartArea.right;
+			y1 = y2 = position;
+		}
+
+		return {
+			x1: x1,
+			x2: x2,
+			y1: y1,
+			y2: y2,
+			width: lineWidth,
+			color: lineColor,
+			borderDash: borderDash,
+			borderDashOffset: borderDashOffset
+		};
+	}
+
+	function collectVisibleGridLines(chart) {
+		var lines = [];
+
+		// Temporary object that stores already collected gridLine positions to prevent gridLines from overlapping
+		var gridLinesHash = {
+			horizontal: {},
+			vertical: {}
+		};
+
+		// Marks scale border positions to prevent overlapping of gridLines and scale borders
+		helpers.each(chart.scales, function(scale) {
+			var scaleOptions = scale.options;
+			var glHashByOrientantion = gridLinesHash[!scale.isHorizontal() ? 'horizontal' : 'vertical'];
+			var borderPosition;
+
+			// gridLines.drawBorder is deprecated
+			if (scaleOptions.gridLines.drawBorder && scaleOptions.borderColor !== null && scaleOptions.borderWidth !== 0 && scaleOptions.borderWidth !== null) {
+				if (scale.isHorizontal()) {
+					borderPosition = scale.position === 'top' ? scale.bottom : scale.top;
+				} else {
+					borderPosition = scale.position === 'left' ? scale.right : scale.left;
+				}
+
+				glHashByOrientantion[Math.round(borderPosition)] = true;
+			}
+		});
+
+		// Collects gridLines
+		helpers.each(chart.scales, function(scale) {
+			var scaleOptions = scale.options;
+			var glHashByOrientantion = gridLinesHash[scale.isHorizontal() ? 'horizontal' : 'vertical'];
+			var position;
+
+			if (scaleOptions.display && scaleOptions.gridLines.display && scaleOptions.gridLines.drawOnChartArea) {
+				for (var tickIndex = 0, ticksCount = scale.ticks.length; tickIndex < ticksCount; tickIndex++) {
+					if (helpers.isNullOrUndef(scale.ticks[tickIndex])) {
```

Comment:
@simonbrunel @benmccann does this check make sense in the context of how ticks changed in 2.7.0?

Reply:
I think so